### PR TITLE
Add faculty verification to user security log

### PR DESCRIPTION
**Pull request by mwharrison**

### CODE REVIEW

**Comment by mwharrison** [comment]
[Write event to Accounts security log when user becomes faculty verified.](https://trello.com/c/uMi1zB3X/474-write-event-to-accounts-security-log-when-user-becomes-faculty-verified)

File: app/routines/update_user_salesforce_info.rb [hunk 1]
```diff
@@ -191,6 +191,13 @@ def cache_contact_data_in_user!(contact, user)
 
       user.faculty_status = case contact.faculty_verified
       when "Confirmed"
+        SecurityLog.create!(
+            user: user.try(:is_anonymous?) ? nil : user,
```
Comment (Dantemss) [inline app/routines/update_user_salesforce_info.rb]: I think the anonymous check is not needed here, the user should never be anonymous if they are getting faculty-verified. You should be able to just pass in the user directly.

File: app/routines/update_user_salesforce_info.rb [hunk 2]
```diff
@@ -191,6 +191,13 @@ def cache_contact_data_in_user!(contact, user)
...
+            application: nil,
+            remote_ip: nil,
+            event_type: :faculty_verified,
+            event_data: user
```
Comment (Dantemss) [inline app/routines/update_user_salesforce_info.rb]: This field should be a json object, not a ruby object. We don't usually pass in a user to event_data (except apparently in a few tests for some reason). I recommend you set the event_data to something like `{ user_id: user.id }`.

File: app/routines/update_user_salesforce_info.rb [hunk 3]
```diff
@@ -191,6 +191,13 @@ def cache_contact_data_in_user!(contact, user)
 
       user.faculty_status = case contact.faculty_verified
       when "Confirmed"
+        SecurityLog.create!(
+            user: user,
+            application: nil,
+            remote_ip: nil,
+            event_type: :faculty_verified,
+            event_data: { user_id: user.id, salesforce_id: contact.id }
```
Comment (mwharrison) [inline app/routines/update_user_salesforce_info.rb]: @Dantemss - what do you think about also having the Salesforce ID, so you can easily go look them up if needed?

Comment (Dantemss) [reply app/routines/update_user_salesforce_info.rb]: That would be great. The more info you can provide the better IMO.

Comment (mwharrison) [reply app/routines/update_user_salesforce_info.rb]: Cool - so does this look ready @Dantemss?

**Pull request merged**